Mathlib/RingTheory/Polynomial/ScaleRoots.lean

```diff
@@ -158,6 +182,72 @@ theorem scaleRoots_aeval_eq_zero_of_aeval_div_eq_zero [Algebra R K]
   scaleRoots_eval₂_eq_zero_of_eval₂_div_eq_zero inj hr hs
 #align polynomial.scale_roots_aeval_eq_zero_of_aeval_div_eq_zero Polynomial.scaleRoots_aeval_eq_zero_of_aeval_div_eq_zero
 
+@[simp]
+lemma scaleRoots_mul (p : R[X]) (r s) :
+    p.scaleRoots (r * s) = (p.scaleRoots r).scaleRoots s := by
+  ext; simp [mul_pow, mul_assoc]
+
+lemma mul_scaleRoots (p q : R[X]) (r : R) :
+    r ^ (natDegree p + natDegree q - natDegree (p * q)) • (p * q).scaleRoots r =
```

Comment:
this deserves a docstring, and I wouldn't be against a `NoZeroDivisors` version of all the below. it took me a while to realise why we need the power of `r`, and I'm honestly not sure where stuff pops up (I think I just need to pick my favourite ring with zero divisors and run some examples to figure that out, though)